src/gflownet/config.py

```diff
@@ -0,0 +1,94 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from omegaconf import MISSING
+
+from gflownet.algo.config import AlgoConfig
+from gflownet.data.config import ReplayConfig
+from gflownet.models.config import ModelConfig
+from gflownet.tasks.config import TasksConfig
+
+
+@dataclass
+class OptimizerConfig:
+    """Generic configuration for optimizers
+
+    Attributes
+    ----------
+    opt : str
+        The optimizer to use (either "adam" or "sgd")
+    learning_rate : float
+        The learning rate
+    lr_decay : float
+        The learning rate decay (in steps, f = 2 ** (-steps / self.cfg.opt.lr_decay))
+    weight_decay : float
+        The L2 weight decay
+    momentum : float
+        The momentum parameter value
+    clip_grad_type : str
+        The type of gradient clipping to use (either "norm" or "value")
+    clip_grad_param : float
+        The parameter for gradient clipping
+    adam_eps : float
+        The epsilon parameter for Adam
+    """
+
+    opt: str = "adam"
+    learning_rate: float = 1e-4
+    lr_decay: float = 20_000
+    weight_decay: float = 1e-8
+    momentum: float = 0.9
+    clip_grad_type: str = "norm"
+    clip_grad_param: float = 10.0
+    adam_eps: float = 1e-8
+
+
+@dataclass
+class Config:
```

Comment:
If you ever need to save/load a config, you could add some utility methods such as `to_{dict|yaml|json}`, `from_{dict|yaml|json}`